### PR TITLE
Add --new-app flag

### PR DESCRIPTION
**Pull request by vogre**
To support non-interactive use from CI

### CODE REVIEW

File: cli/src/commands/deploy.ts [hunk 1]
```diff
@@ -194,7 +198,10 @@ export default class Deploy extends Command {
   }
 
   public async run() {
-    const { flags: { 'app-name': givenAppName, env: givenEnv } } = this.parse(Deploy);
+    const { flags: { 'app-name': givenAppName, env: givenEnv, 'new-app': forceNewApp } } = this.parse(Deploy);
+    if (givenAppName && forceNewApp) {
+      this.error('--app-name and --new-app flags are incompatible');
```
Comment (arielshaqed) [inline cli/src/commands/deploy.ts]: Not sure why, actually?

File: cli/src/commands/deploy.ts [hunk 2]
```diff
@@ -220,10 +227,10 @@ export default class Deploy extends Command {
 
     this.startStage('register app on local');
     const env = 'default'; // hardcoded for now
-    let project = findProjectByDirectory(projects, projectDir);
+    let project = forceNewApp ? undefined : findProjectByDirectory(projects, projectDir);
 
     let application: Application;
-    if (!project || givenAppName) {
+    if (!forceNewApp && (!project || givenAppName)) {
```
Comment (arielshaqed) [inline cli/src/commands/deploy.ts]: This logic is getting _very_ hairy.  In particular, `forceNewApp` only ever appears in the negative... which makes for interesting reading.

At a micro level, I *think* this conditional might be easier to read in its negated dual form:
```suggestion
    if (!(forceNewApp || (project && !givenAppName)() {
```

File: cli/src/test/deploy.spec.ts [hunk 3]
```diff
@@ -120,6 +120,15 @@ test('given --app-name but app not in list gives informative message', async (t)
   t.snapshot({ ...result, out: result.out.replace(t.context.projectDir, 'PROJECT_DIR') });
 });
 
+test('project associated deploys to new app with --new-app', async (t) => {
+  const newApp = createApp({ id: 'hybrid-koala' }); // hybrid-koala != fluffy-samaritan (default)
+  td.when(t.context.uploadFake()).thenReturn({ ok: true, digest: 'abcabc' });
+  td.when(t.context.lycanFake.deployInitial(anything, 'default', 'abcabc', [])).thenResolve(newApp);
+  const result = await t.context.shell.run(`${t.context.run} deploy --new-app`, 'utf-8');
+  t.snapshot({ ...result, out: result.out.replace(t.context.projectDir, 'PROJECT_DIR') });
```
Comment (arielshaqed) [inline cli/src/test/deploy.spec.ts]: Not sure this entirely tests  the desired "new app" behaviour.

File: cli/src/test/deploy.spec.ts [hunk 4]
```diff
@@ -120,6 +120,15 @@ test('given --app-name but app not in list gives informative message', async (t)
   t.snapshot({ ...result, out: result.out.replace(t.context.projectDir, 'PROJECT_DIR') });
 });
 
+test('project associated deploys to new app with --new-app', async (t) => {
+  const newApp = createApp({ id: 'hybrid-koala' }); // hybrid-koala != fluffy-samaritan (default)
```
Comment (arielshaqed) [inline cli/src/test/deploy.spec.ts]: What would happen if you passed `fluffy-samaritan`?  Should that also be a test??

File: cli/src/commands/deploy.ts [hunk 5]
```diff
@@ -194,7 +198,10 @@ export default class Deploy extends Command {
   }
 
   public async run() {
-    const { flags: { 'app-name': givenAppName, env: givenEnv } } = this.parse(Deploy);
+    const { flags: { 'app-name': givenAppName, env: givenEnv, 'new-app': forceNewApp } } = this.parse(Deploy);
```
Comment (arielshaqed) [inline cli/src/commands/deploy.ts]: It seems OClif can do this.  Perhaps use `exclusive` (https://oclif.io/docs/flags)?

File: cli/src/commands/deploy.ts [hunk 6]
```diff
@@ -218,40 +225,43 @@ export default class Deploy extends Command {
...
-          directory: projectDir,
-          applicationId,
-          defaultEnv: env,
-        };
-        // Don't update association if given --app-name option
-        if (!givenAppName) {
-          projects.push(project);
-          this.conf.set('projects', projects);
-        }
+        this.startStage('register app on local');
```
Comment (arielshaqed) [inline cli/src/commands/deploy.ts]: 1. This times local execution.
2. Having it conditional will make analytics harder.  Let's either not have a stage, or just `this.startStage('start')` before starting.

**Comment by arikmaor** [comment]
> To support non-interactive use from CI

We need to have an API key support for the CI...

**Comment by vogre** [comment]
> We need to have an API key support for the CI...

We currently do something like 
```bash
export RESHUFFLE_CONFIG=/path/to/reshuffle.config.yml
echo accessToken: $RESHUFFLE_ACCESS_TOKEN > $RESHUFFLE_CONFIG
```

**Pull request merged**